Unpin twine in publish step

We have pinned twine due to an error it raises in the latest version;

```
Installing collected packages: rfc3986, requests-toolbelt, nh3, more-itertools, jeepney, jaraco.context, id, docutils, SecretStorage, readme-renderer, jaraco.functools, jaraco.classes, keyring, twine
Successfully installed SecretStorage-3.3.3 docutils-0.21.2 id-1.5.0 jaraco.classes-3.4.0 jaraco.context-6.0.1 jaraco.functools-4.1.0 jeepney-0.9.0 keyring-25.6.0 more-itertools-10.6.0 nh3-0.2.21 readme-renderer-44.0 requests-toolbelt-1.0.0 rfc3986-2.0.0 twine-6.1.0
Checking 
dist/carolina-1.0.16.dev5+g2cf9d21-cp310-cp310-manylinux_2_17_x86_64.manylinux20
14_x86_64.whl: ERROR    InvalidDistribution: Invalid distribution metadata: unrecognized or    
         malformed field 'license-file'                                         
Error: Process completed with exit code 1.
```

Ref.
https://github.com/equinor/Carolina/actions/runs/13786242189/job/38554987259#step:7:94

--- a/.github/workflows/bundle_with_dakota_caller.yml
+++ b/.github/workflows/bundle_with_dakota_caller.yml
@@ -102,7 +102,7 @@ jobs:
       - name: Validate dist with twine check
         run: |
           ls dist
-          pip install twine==6.0.1
+          pip install twine
           twine check dist/*
 
           # remove carolina dev-wheels if any

--- a/.github/workflows/bundle_with_dakota_caller.yml
+++ b/.github/workflows/bundle_with_dakota_caller.yml
@@ -114,7 +114,7 @@ jobs:
           ls dist
 
       - name: Publish to pypi
-        uses: pypa/gh-action-pypi-publish@v1.8.14
+        uses: pypa/gh-action-pypi-publish@v1.12.4
         if: >
           (github.event_name == 'workflow_dispatch' && inputs.publish) ||
           (github.event_name == 'push' && startsWith(github.ref, 'refs/tags'))

--- a/.github/workflows/bundle_with_dakota_caller.yml
+++ b/.github/workflows/bundle_with_dakota_caller.yml
@@ -103,6 +103,7 @@ jobs:
         run: |
           ls dist
           pip install twine
+          pip install packaging==25.0
           twine check dist/*
 
           # remove carolina dev-wheels if any

